doc/arch/014-reporting-etl-pipeline.md

```diff
@@ -0,0 +1,150 @@
+# Reporting ETL Pipeline
+
+## Context
+
+Simple's dashboard reports are currently generated by queries on the application database's tables. This has several
+disadvantages as we start to scale.
+
+### 1. Complexity
+
+Simple's application schema is optimized for application transactions like registering patients, recording BPs,
+updating patient info, or tracking overdue patients. As such, reporting queries grow complex. Asking questions like
+"latest BP in a month", "prescribed drugs as of a date", "current step of treatment protocol" involve complex clauses
+in the SQL queries.
```

Comment:
> involve complex clauses in the SQL queries.

I would phrase this as: involve joining across multiple tables and using significant amount of ruby code to transform the data into reports.

While some queries are complex, it is the amount of ruby code in my opinion that amounts to complexity. Further, we're actually adding significantly complex queries to create these schemas :) So we're actually not shying away from complex SQL queries no? We want overall report creation: sql + ruby to be simpler.